src/Private/Helpers.ps1

```diff
@@ -3775,4 +3732,75 @@ function Copy-PodeObjectDeepClone {
         # Deserialize the XML back into a new PSObject, creating a deep clone of the original
         return [System.Management.Automation.PSSerializer]::Deserialize($xmlSerializer)
     }
+}
+
+<#
+.SYNOPSIS
+    Converts a hashtable to a ConcurrentDictionary.
+
+.DESCRIPTION
+    The `ConvertTo-PodeConcurrentStructure` function takes a hashtable and converts it into a
+    ConcurrentDictionary, which provides thread-safe operations for adding and retrieving items.
+    This function supports the recursive conversion of nested hashtables, ordered dictionaries, and arrays.
+
+.PARAMETER InputObject
+    The hashtable to be converted into a ConcurrentDictionary.
+
+.OUTPUTS
+    [System.Collections.Concurrent.ConcurrentDictionary[string, object]]
+    Returns a ConcurrentDictionary with the same keys and values as the input hashtable.
+
+.EXAMPLE
+    $hashTable = @{
+        Key1 = 'Value1'
+        Key2 = @{
+            SubKey1 = 'SubValue1'
+        }
+    }
+    $concurrentDictionary = ConvertTo-PodeConcurrentStructure -InputObject $hashTable
+    # The variable $concurrentDictionary now contains a ConcurrentDictionary with the same structure as $hashTable
+
+.NOTES
+    This is an internal function and may change in future releases of Pode.
+#>
+function ConvertTo-PodeConcurrentStructure {
+    param (
+        [object]
+        $InputObject
+    )
+
+    if ($InputObject -is [hashtable]) {
+        $concurrentDictionary = [System.Collections.Concurrent.ConcurrentDictionary[string, object]]::new([StringComparer]::OrdinalIgnoreCase)
+    }
+    elseif ($InputObject -is [System.Collections.Specialized.OrderedDictionary]) {
+        $concurrentDictionary = [Pode.PodeOrderedConcurrentDictionary[string, object]]::new([StringComparer]::OrdinalIgnoreCase)
+    }
+    elseif ($InputObject -is [System.Object[]]) {
+        $array = [System.Collections.ArrayList]::Synchronized([System.Collections.ArrayList]::new())
+        foreach ($item in $InputObject) {
+            # Convert each item and add to the synchronized array
+            $convertedItem = ConvertTo-PodeConcurrentStructure -InputObject $item
+            [void]$array.Add($convertedItem)
+        }
+
+        # Return synchronized ArrayList without unrolling
+        return , $array
+    }
+    else {
+        # If the object is neither a hashtable, ordered dictionary, nor array, return it as-is
+        return $InputObject
```

Comment:
Does this mean we should always cast to  `ConvertTo-PodeConcurrentStrucutre`? That may reduce the cognitive overhead and just assume that we always cast. Maybe add another check at the top to return something that's already been cast to keep the cost low.

Reply:
I created this function primarly for https://github.com/Badgerati/Pode/pull/1391 (PodeState threadsafe)
The reason of this function is to convert any hashtable or array that has to be thread safe to as safe-thread object 
System.Collections.Concurrent.ConcurrentDictionary for hashtable and System.Collections for array.

It's not meant to be used for any hashtable or array